symmecrypt.go

```diff
@@ -27,9 +50,8 @@ type Key interface {
 
 // A KeyFactory instantiates a Key
 type KeyFactory interface {
-	NewKey(string) (Key, error)
+	NewKey(*KeyConfig) (Key, error)
```

Comment:
don't move KeyConfig out of keyload
it's better to add `NewSequentialKey` in the `KeyFactory` interface